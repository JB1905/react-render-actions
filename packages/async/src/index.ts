export { Async } from './Async';
export { Await } from './Await';
export { Try } from './Try';
export { Catch } from './Catch';
export { Finally } from './Finally';
