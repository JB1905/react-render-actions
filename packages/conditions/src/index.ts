export { If } from './If';
export { Else } from './Else';
export { Then } from './Then';
export { When } from './When';
export { Unless } from './Unless';
