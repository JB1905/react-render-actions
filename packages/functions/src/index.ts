export { Map } from './Map';
export { Filter } from './Filter';
export { Reduce } from './Reduce';
