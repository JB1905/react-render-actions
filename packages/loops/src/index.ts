export { For } from './For';
export { ForOf } from './ForOf';
export { ForIn } from './ForIn';
export { ForEach } from './ForEach';
export { While } from './While';
export { DoWhile } from './DoWhile';
export { Break } from './Break';
export { Continue } from './Continue';
