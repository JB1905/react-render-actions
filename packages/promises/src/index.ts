export { RPromise } from './Promise';
export { Resolve } from './Resolve';
export { Reject } from './Reject';
