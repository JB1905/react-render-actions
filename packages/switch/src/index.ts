export { Switch } from './Switch';
export { Case } from './Case';
export { Default } from './Default';
