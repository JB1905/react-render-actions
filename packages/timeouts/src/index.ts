export { SetTimeout } from './SetTimeout';
export { SetInterval } from './SetInterval';
