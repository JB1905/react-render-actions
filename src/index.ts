export { If } from './Conditions/If';
export { Else } from './Conditions/Else';
export { Then } from './Conditions/Then';
export { When } from './Conditions/When';
export { Unless } from './Conditions/Unless';

export { Map } from './Functions/Map';
export { Filter } from './Functions/Filter';
export { Reduce } from './Functions/Reduce';

export { For } from './Loops/For';
export { ForOf } from './Loops/ForOf';
export { ForIn } from './Loops/ForIn';
export { ForEach } from './Loops/ForEach';
export { While } from './Loops/While';
export { DoWhile } from './Loops/DoWhile';
export { Break } from './Loops/Break';
export { Continue } from './Loops/Continue';

export { RPromise } from './Promises/Promise';
export { Resolve } from './Promises/Resolve';
export { Reject } from './Promises/Reject';

export { Switch } from './Switch/Switch';
export { Case } from './Switch/Case';
export { Default } from './Switch/Default';

export { SetTimeout } from './Timeouts/SetTimeout';
export { SetInterval } from './Timeouts/SetInterval';
