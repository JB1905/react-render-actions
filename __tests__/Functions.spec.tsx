import React from 'react';
import { render, getAllByTestId } from '@testing-library/react';

import { Map, Filter, Reduce } from '../src';

const bands = [
  { name: 'The Killers' },
  { name: 'Green Day' },
  { name: 'The Offspring' },
  { name: 'Sum 41' },
  { name: 'Blink 182' },
  { name: 'U2' },
];

interface Output {
  name: string;
}

describe('Map', () => {
  it('should map provided values', () => {
    const { container, getByText, getAllByTestId } = render(
      <Map data={bands}>
        {({ name }: Output) => (
          <p data-testid="band-name" key={name}>
            {name}
          </p>
        )}
      </Map>
    );

    expect(container).toMatchSnapshot();

    bands.map(({ name }) => expect(getByText(name)).toBeDefined());

    expect(getAllByTestId('band-name').length).toBe(6);
  });
});

describe('Filter', () => {
  it('should filter provided values', () => {
    const pattern = ({ name }: Output) => name !== 'U2';

    const { container, getByTestId } = render(
      <Filter data={bands} pattern={pattern}>
        {(filteredData: Output[]) => (
          <p data-testid="result">{JSON.stringify(filteredData)}</p>
        )}
      </Filter>
    );

    expect(container).toMatchSnapshot();

    const resultContent = getByTestId('result').innerHTML;

    expect(resultContent).toBe(JSON.stringify(bands.filter(pattern)));

    expect(JSON.parse(resultContent).length).toBe(5);
  });
});

describe('Reduce', () => {
  it('should reduce provided values', () => {
    const data = [1, 2, 3, 4, 5];

    const pattern = (prev: number, next: number) => prev + next;

    const { container, getByTestId } = render(
      <Reduce data={data} pattern={pattern}>
        {(value: number) => <p data-testid="result">{value}</p>}
      </Reduce>
    );

    expect(container).toMatchSnapshot();

    expect(getByTestId('result').innerHTML).toBe(
      data.reduce(pattern).toString()
    );
  });
});
