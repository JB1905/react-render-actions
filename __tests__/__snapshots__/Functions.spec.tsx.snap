// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should filter provided values 1`] = `
<div>
  <p
    data-testid="result"
  >
    [{"name":"The Killers"},{"name":"Green Day"},{"name":"The Offspring"},{"name":"Sum 41"},{"name":"Blink 182"}]
  </p>
</div>
`;

exports[`Map should map provided values 1`] = `
<div>
  <p
    data-testid="band-name"
  >
    The Killers
  </p>
  <p
    data-testid="band-name"
  >
    Green Day
  </p>
  <p
    data-testid="band-name"
  >
    The Offspring
  </p>
  <p
    data-testid="band-name"
  >
    Sum 41
  </p>
  <p
    data-testid="band-name"
  >
    Blink 182
  </p>
  <p
    data-testid="band-name"
  >
    U2
  </p>
</div>
`;

exports[`Reduce should reduce provided values 1`] = `
<div>
  <p
    data-testid="result"
  >
    15
  </p>
</div>
`;
